catppuccinpalette.dtx.tera

```diff
@@ -0,0 +1,362 @@
+% \columnbreak
+% \showCol{Ctp{{ id | capitalize }}Success}\par
+% \showCol{Ctp{{ id | capitalize }}Warning}\par
+% \showCol{Ctp{{ id | capitalize }}Error}\par
+% \end{multicols}
+{% endfor %}
+%
+% \StopEventually{}
+%
+% \section{Implementation}
+%
+% \iffalse
+%<*package>
+% \fi
+%
+%    \begin{macrocode}
+\NeedsTeXFormat{LaTeX2e}[1994/06/01]
+\RequirePackage{pgfopts}
+\RequirePackage{xcolor}
+%    \end{macrocode}
+% Define the conditionals used by package options
+%    \begin{macrocode}
+\newif\ifcatppuccinpalette@pagecolor
+\newif\ifcatppuccinpalette@textcolor
+\pgfkeys{
+	/catppuccinPalette/.cd,
+%    \end{macrocode}
+% Define the package option keys
+%    \begin{macrocode}
+	Latte/.style={style=Latte},
+	latte/.style={style=Latte},
+	Frappe/.style={style=Frappe},
+	frappe/.style={style=Frappe},
+	Macchiato/.style={style=Macchiato},
+	macchiato/.style={style=Macchiato},
+	Mocha/.style={style=Mocha},
+	mocha/.style={style=Mocha},
+	style/.is choice,
+	style/.value required,
+	{{ 'style/Latte/.code = {\def\catppuccinpalette@style{Latte}},' }}
+	style/latte/.style = {style=Latte},
+	{{ 'style/Frappe/.code = {\def\catppuccinpalette@style{Frappe}},' }}
+	style/frappe/.style = {style=Frappe},
+	{{ 'style/Macchiato/.code = {\def\catppuccinpalette@style{Macchiato}},' }}
+	style/macchiato/.style = {style=Macchiato},
+	{{ 'style/Mocha/.code = {\def\catppuccinpalette@style{Mocha}},' }}
+	style/mocha/.style = {style=Mocha},
+	textcolor/.is if = catppuccinpalette@textcolor,
+	textcolor/.default = true,
+	pagecolor/.is if = catppuccinpalette@pagecolor,
+	pagecolor/.default = true,
+	styleAll/.style = {pagecolor,textcolor},
+%    \end{macrocode}
+% Set initial values for all options
+%    \begin{macrocode}
+	textcolor = false,
+	pagecolor = false,
+	style = Latte,
+}
+\ProcessPgfOptions{catppuccinPalette}
+%    \end{macrocode}
+{% for id, flavor in flavors %}
+{{ '% Define the \emph{' }}{{ id | capitalize }}{{ '} colorset' }}
+%    \begin{macrocode}
+{{ '\preparecolorset{RGB/HTML}{Cat' }}{{ id | capitalize }}{{ '}{}{%' }}{% for name,color in flavor.colors %}
+	{{ name | capitalize }},{{ color.rgb.r }},{{ color.rgb.g }},{{ color.rgb.b }}/{{ color.hex | upper }}{% if not loop.last %};{% endif %}%{% endfor %}
+}
+{{ '\colorlet{Cat' }}{{ id | capitalize }}{{ 'Background}{Cat' }}{{ id | capitalize }}{{ 'Base}' }}
+{{ '\colorlet{Cat' }}{{ id | capitalize }}{{ 'Link}{Cat' }}{{ id | capitalize }}{{ 'Blue}' }}
+{{ '\colorlet{Cat' }}{{ id | capitalize }}{{ 'Success}{Cat' }}{{ id | capitalize }}{{ 'Green}' }}
+{{ '\colorlet{Cat' }}{{ id | capitalize }}{{ 'Warning}{Cat' }}{{ id | capitalize }}{{ 'Yellow}' }}
+{{ '\colorlet{Cat' }}{{ id | capitalize }}{{ 'Error}{Cat' }}{{ id | capitalize }}{{ 'Red}' }}
+
+{{ '\preparecolorset{RGB/HTML}{Ctp' }}{{ id | capitalize }}{{ '}{}{%' }}{% for name,color in flavor.colors %}
+	{{ name | capitalize }},{{ color.rgb.r }},{{ color.rgb.g }},{{ color.rgb.b }}/{{ color.hex | upper }}{% if not loop.last %};{% endif %}%{% endfor %}
```

Comment:
```suggestion
{{ '\preparecolorset{RGB/HTML}{Ctp' }}{{ id | capitalize }}{{ '}{}{%' }}{% for id,color in flavor.colors %}
	{{ id | capitalize }},{{ color.rgb.r }},{{ color.rgb.g }},{{ color.rgb.b }}/{{ color.hex | upper }}{% if not loop.last %};{% endif %}%{% endfor %}
```
The `identifier` and `name` properties should be named accordingly for clarity - `flavor.name` would be like Frappé, `flavor.identifier`/`id` would be like `frappe`.

Reply:
But it's not `flavor.name` but the key in the `flavor.colors` map I'm referring to here. I know it's hard to read but at least for now I tried to avoid spurious newlines.

Or am I reading this incorrectly?

Reply:
The key is the identifier yup.

Reply:
Hm I think we still do not talk of the same thing.

So it is
```tex
{% for id, flavor in flavors %}
{{ '% Define the \emph{' }}{{ id | capitalize }}{{ '} colorset' }}
%    \begin{macrocode}
{{ '\preparecolorset{RGB/HTML}{Cat' }}{{ id | capitalize }}{{ '}{}{%' }}
{% for name,color in flavor.colors %}
	{{ name | capitalize }},{{ color.rgb.r }},{{ color.rgb.g }},{{ color.rgb.b }}/{{ color.hex | upper }}{% if not loop.last %};{% endif %}%
{% endfor %}
...
{% endfor %}
```

so by your change you'd replace the name of the color by the name of the flavor.

Reply:
Ah I see you already have the `id` set. I would do this then:
```suggestion
{{ '\preparecolorset{RGB/HTML}{Ctp' }}{{ id | capitalize }}{{ '}{}{%' }}{% for _,color in flavor.colors %}
	{{ color.identifier | capitalize }},{{ color.rgb.r }},{{ color.rgb.g }},{{ color.rgb.b }}/{{ color.hex | upper }}{% if not loop.last %};{% endif %}%{% endfor %}
```

Reply:
Ah yes that might be easier to read. Maybe doing the same with `id` for the flavor would also be better.